harness/determined/keras/_tf_keras_trial.py

```diff
@@ -900,8 +905,10 @@ def _post_train_batch_end(self, num_inputs: int, logs: Dict) -> None:
             )
 
         if self.context.distributed.size > 1:
-            num_inputs = self._hvd_allreduce(num_inputs, average=False, name="train_num_inputs")
-            num_inputs = self._convert_possible_tensor(num_inputs)
+            self.train_workload_inputs = self._hvd_allreduce(
```

Comment:
if we start supporting native dtrain backend, should we replace this call with torch.distributed primitive?  or should we add a zmq allreduce?

Reply:
this is ... keras?  I don't expect TFKerasTrial to work with non-horovod ever

Reply:
i thought there was something similar in pytorch but all i see is
`hvd.broadcast_object(metrics...)`
which we can replace with zmq as part of native dtrain probably.